src/components/highlights/Highlights.ts

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {Highlighter} from 'core';
+import {showViewer} from 'store/widgets/actions';
+import { mousePosition } from 'common/dom';
+
+interface IHighlightsProps {
+  annotations: any[];
+  showViewer: (x: number, y: number, annotationId: number) => void;
+}
+
+@connect(
+  state => ({
+    annotations: state.annotations.data,
+  }),
+  dispatch => ({
+    showViewer: (x, y, annotationId) => dispatch(showViewer(x, y, annotationId)),
+  }),
+)
+export default class Highlights extends React.Component<Partial<IHighlightsProps>, {}> {
```

Comment:
I wrapped this logic in a React component just to comply with Redux philosophy (and more prosaically, to use `connect`). I guess we can also use `connect` more directly or connect to the store in any other way. The important point was to render annotations and rerender them if they change.

Reply:
I think [store.subscribe](http://devdocs.io/redux/api/store#subscribe) would me perfectly suitable for this scenario.

it could be handled in a plain ts module(file) imported and initialized in `bootstrap.ts`, for example (or if short enough, directly in `bootstrap.ts`)

Reply:
Moved to roadmap (`miscalleneous`)